samples_project/Assets/SampleViewer/Samples/Measure/Measure.cs

```diff
@@ -0,0 +1,298 @@
+﻿// Copyright 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+
+using System;
+using System.Collections.Generic;
+using UnityEngine.UI;
+using Esri.ArcGISMapsSDK.Components;
+using Esri.ArcGISMapsSDK.Utils.GeoCoord;
+using Esri.GameEngine.Geometry;
+using Esri.HPFramework;
+using UnityEngine;
+using Unity.Mathematics;
+
+public enum UnitType
+{
+    m = 0,
+    km = 1,
+    mi = 2,
+    ft=3
+}
+
+public class Measure : MonoBehaviour
+{
+    public GameObject Line;
+    public Text GeodedicDistanceText;
+    private String unitTxt;
+    public GameObject LineMarker;
+    public GameObject InterpolationMarker;
+    public float InterpolationInterval=100;
+    public Dropdown UnitDropdown;
+    public Button ClearButton;
+    private HPRoot hpRoot;
+    private ArcGISMapComponent arcGISMapComponent;
+    private float elevationOffset = 5f;
+    private GameObject FeaturePoint;
+    private List<GameObject> featurePoints = new List<GameObject>();
+    private Stack<GameObject> stops = new Stack<GameObject>();
+    private GameObject lastStop;
+    private ArcGISLocationComponent lastStopLocation;
+    private double3 lastRootPosition;
+    private ArcGISPoint thisPoint;
+    private ArcGISPoint lastPoint;
+    private ArcGISPoint prePoint;
+    private ArcGISPoint nextPoint;
+    private double geodedicDistance=0;
+    private LineRenderer lineRenderer;
+    private ArcGISSpatialReference spatialRef = new ArcGISSpatialReference(3857);
```

Comment:
I would remove this hardcoded spatial reference. You can just get the spatial reference from the map component.